docs: explain how to configure the dagger engine

TomChv opened the pull request:
Explain how to configure mirror on the dagger engine

Resolves: #5055

gerhard reviewed (commented):
LGTM!

Want to approve & merge @vikram-dagger?

cc @shykes

vikram-dagger commented on docs/current/guides/967042-customize-engine-configuration.md:
```diff
@@ -0,0 +1,53 @@
+
+## Introduction
+
+This guide explains how to configure Dagger Engine through the custom `engine.toml` file.
```
```suggestion
This guide explains how to configure the Dagger Engine through the custom `engine.toml` file.
```

vikram-dagger commented on docs/current/guides/967042-customize-engine-configuration.md:
```diff
@@ -0,0 +1,53 @@
+This guide explains how to configure Dagger Engine through the custom `engine.toml` file.
+
+For example, this can be used to configure a registry mirror for pulling container images instead of DockerHub.
+
```
Can you please add a "requirements" section here? similar to https://docs.dagger.io/882813/build-test-publish-java-spring#requirements, updating as needed.

vikram-dagger replied:
Added

vikram-dagger commented on docs/current/guides/967042-customize-engine-configuration.md:
```diff
@@ -0,0 +1,53 @@
+
+### Registry mirror
+
+Create a file `engine.toml` that contains the registry mirror, e.g.:
```
I suggest renumbering this as a list of steps.

vikram-dagger reviewed (commented):
See general feedback. Once you review and update, I'll do a copy edit on the text.

TomChv commented on docs/current/guides/967042-customize-engine-configuration.md:
```diff
@@ -0,0 +1,53 @@
+## Conclusion
+
+This guide described one way to customize the Dagger Engine configuration.
+Learn how to use [alternative runtimes for Dagger Engine](541047-alternative-runtimes.md), e.g. Podman, Kubernetes, etc.
```
I meant if you deploy the dagger engine in a Kubernetes cluster

TomChv commented on docs/current/guides/967042-customize-engine-configuration.md:
````diff
@@ -0,0 +1,53 @@
+export _EXPERIMENTAL_DAGGER_RUNNER_HOST=docker-container://customized-dagger-engine
+dagger query --progress=plain <<< '{ container { from(address:"hello-world") { stdout } } }'
+```
+
````
How? If you export this env variable, the customized engine can only be used

vikram-dagger replied:
Added

vikram-dagger commented:
@TomChv, please see my changes and merge if OK.

vikram-dagger commented on docs/current/guides/967042-modify-engine-configuration.md:
````diff
@@ -0,0 +1,56 @@
+  dagger query --progress=plain <<< '{ container { from(address:"hello-world") { stdout } } }'
+  ```
+
+  You should see the specified `hello-world` container being pulled from the mirror instead of from Docker Hub.
````
@TomChv I added this but please confirm this is what should happen or modify if wrong.

sipsma commented:
I'm not totally sure if it's a good idea to put this in our "official website docs". The reason it's just in the "advanced" operator manual on our repo is that it's currently a pretty advanced use case that requires custom setup of Dagger, use of `_EXPERIMENTAL_*` env vars etc. We've generally tried to avoid going into all of that on our official docs since it's all meant to be a temporary solution to unblock power users, with better interfaces to come later.

For example, this guide is currently highly specific to a custom engine running in docker, but most of the time power users are running somewhere else, so it's not clear if the docker-specific instructions would be all that useful to them; I fear it may just raise more questions than it answers in addition to enshrining our current hacky OX as the "official" way.

@shykes curious to get your thoughts here

TomChv commented:
@sipsma Should we put it into the operator manual then?

TomChv commented:
@vikram-dagger @sipsma I updated this PR by removing the guide and instead added a small line in the FAQ that redirect to our operator guide.
I think it's still important to mention it on our doc so user can find a result if they use the search bar but same as rootless, most of the content is in the operator guide.

Let me know if there's anything to change.

@vikram-dagger This change lighten the complete exemple but we have no guided example in our operator manual, so I just added a small one line example to show how to replace the default config file with the user's one.

TomChv commented on docs/current/faq.md:
```diff
@@ -170,6 +170,10 @@ Operations are automatically cached every time a Dagger pipeline runs. [Cache vo
 
 No. "Rootless mode" means running the Dagger Engine as a container without the `--privileged` flag. In this case, the container would not run as the "root" user of the system. Currently, the Dagger Engine cannot be run as a rootless container; network and filesystem constraints related to rootless usage would significantly limit its capabilities and performance. [Read more about these constraints ...
 
+### Can I configure the Dagger Engine Runner?
```
That's right, I'll update it today

sipsma reviewed (approved):
One nit I think is worth addressing, otherwise LGTM.
